README.md

```diff
@@ -1,8 +1,102 @@
+    ```bash
+    ./gradlew systemTest --tests="com.google.cloud.dataproc.jdbc.DataprocSystemTest" -DprojectId="{projectId}" -Dregion="{region}" -Duser="{user}"
+    ```
+    Note that projectId and region are the parameters for the specific project you are working on; user can be an arbitrary username, used as a prefix for the cluster name to create, delete and avoid cluster conflict during testing.
+
+## How to use Dataproc Driver
+1. Clone this repo
+    ```bash
+    git clone https://github.com/GoogleCloudDataproc/dataproc-jdbc-connector.git
+    cd dataproc-jdbc-connector
+    ``` 
+2. Build Dataproc Driver JAR
+    ```bash
+    ./gradlew -p jdbc-driver shadowJar
+    ```
+    Note that this step might take around 45 minutes. <br />
+    Compiled Dataproc Driver JAR will be at `dataproc-jdbc-connector/jdbc-driver/build/libs/jdbc-driver-1.0-SNAPSHOT-all.jar`.
+
+2. Build example-client JAR
+    ```bash
+    ./gradlew -p example-client shadowJar
+    ```
+
+### Connection URL format
+* Dataproc Driver for Hive accepts JDBC URL string with prefix `jdbc:dataproc://hive/`
+* Client can pass in database name or leave it as empty to use default database <br>
+  `jdbc:dataproc://hive/;` or `jdbc:dataproc://hive/dbName;`
+* `projectId` and `region` are required parameters. To find your project and region, refer back to [Set up a Cloud project](https://cloud.google.com/dataproc/docs/guides/setup-project) in the **Before you begin** section above. <br>
+  `jdbc:dataproc://hive/;projectId={pid};region={region}`
+* `clusterName` is an optional parameter that allows client to specify the name of the cluster to connect to. <br>
+    `jdbc:dataproc://hive/;projectId={pid};region={region};clusterName={cluster-name}`
+* `clusterPoolLabel` is an optional parameter that supports submitting to a cluster pool. Client can pass in cluster pool labels and the Dataproc Driver will pick a healthy cluster (cluster with `status.state = ACTIVE`) within the pool to connect to. Please do not put `status.state` as one of the cluster pool label. 
+ 
+    Labels can be specified in the format `clusterPoolLabel=label1=value1[:label2=value2]`
+       
+    Example: 
+    ```bash
+    jdbc:dataproc://hive/;projectId={pid};region={region};clusteroPoolLabel=com=google:team=dataproc`
+    ```
+    Note that `clusterName` has a higher priority than `clusterPoolLabel`, if you pass in both parameters, the DataprocDriver will only look for the cluster by name.
+* DataprocDriver also accepts other semicolon separated list of session variables, Hive configuration variables or Hive variables that [Hive supports](https://cwiki.apache.org/confluence/display/Hive/HiveServer2+Clients#HiveServer2Clients-ConnectionURLFormat)
+
+    ```bash
+    jdbc:dataproc://hive/{dbName};projectId={pid};region={region};clusterName={name};sess_var_list?hive_conf_list#hive_var_list
+    ```
+    
+### Use with other JDBC client
```

Comment:
The word "other" here is confusing because you haven't brought up a first client before this :) 

Maybe just name this section `Using Beeline` for now?